### PR TITLE
Support preferences and override in send method

### PR DESCRIPTION
**Pull request by aydrian**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -19,7 +19,9 @@ const send = (options: ICourierClientConfiguration) => {
       data: params.data,
       event: params.eventId,
       profile: params.profile,
-      recipient: params.recipientId
+      recipient: params.recipientId,
+      preferences: params.preferences,
```
Comment (DannyDouglass) [inline src/client.ts]: build is failing here: `The key 'preferences' is not sorted alphabetically`

Comment (aydrian) [reply src/client.ts]: Interesting. I wondered why they were in that order.

Comment (DannyDouglass) [reply src/client.ts]: some of us use `tslint` to automatically show these issues in vscode if you are interested: https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin

Comment (aydrian) [reply src/client.ts]: Thank you. That helps.

**Pull request merged**